lib/services/database.go

```diff
@@ -200,6 +195,40 @@ func ValidateDatabase(db types.Database) error {
 	return nil
 }
 
+// validateMongoDB validates MongoDB URIs with "mongodb" schemes.
+func validateMongoDB(db types.Database) error {
+	connString, err := connstring.ParseAndValidate(db.GetURI())
+	if err != nil {
+		// connstring.ParseAndValidate requires DNS resolution on TXT/SRV
+		// records for a full validation for "mongodb+srv" URIs. Log a warning
+		// here when DNS resolution fails as we mostly care if the URI is
+		// "lexically" correct at this point. Full validation is more
+		// approriate for health checks or new connections.
+		if isDNSError(err) {
+			logrus.Warnf("MongoDB database %q (connection string %q) failed validation with DNS error: %v", db.GetName(), db.GetURI(), err)
+			return nil
```

Comment:
If we hit DNS error and return here, we won't do any of the checks below, right? So an incorrect read preference for example can sneak in? I assume the returned connString is empty - is there a way to still do the other checks?

Reply:
`connString` will not be empty but without reading driver code, it's hard to tell what has been parsed and what's not when this DNS error is returned.

> is there a way to still do the other checks?
We have to fork mongo-driver and add/expose some extra options for parsing.

Reply:
Looking at the driver's code, it only tries to use DNS resolver if the scheme is `mongodb+srv://`. What if we change the scheme to `mongodb://` just for the sake of validation? A bit hacky but might work?

Reply:
i think it will work but the driver does some extra validations for `mongodb+srv://` type like `mongodb+srv://` URI must have no port, or `srv` specific parameters like `srvmaxhosts` etc. so we will lose those if we replace the scheme.

the `dnsResolver` is an interface so ideally if we can just pass in a fake for this case, but driver doesn't provide that option. we can technically overwrite the `DefaultResolver` from `"go.mongodb.org/mongo-driver/x/mongo/driver/dns"` but there could be race condition if someone is actually making a mongo connection at the same time.

Reply:
maybe we only replace scheme when we run into the DNS error and `ParseAndValidate` again withe the replaced URI?

Reply:
Yep that sounds like a good idea.